### PR TITLE
feat(headless): follow up endpoint created in the api client

### PR DESCRIPTION
**Pull request by mmitiche**
## Summary
Implements the streaming infrastructure and API endpoint for follow-up answers, enabling streaming of follow-up responses from the backend.

### Streaming Strategy
- follow-up-answer-strategy.ts: Handles SSE stream processing for follow-up answers
  - Manages stream lifecycle (open, message, error, close)
  - Dispatches Redux actions for header, text chunks, citations, completion, and errors
  - Tracks answerId via closure for proper state targeting
- follow-up-answer-strategy.test.ts - Tests all stream event handlers and edge cases

### API Endpoint
- follow-up-endpoint.ts - RTK Query endpoint for follow-up generation
     - Parameters: `q`, `conversationId`, `searchHub`, `pipeline`, `pipelineRuleParameters`

- follow-up-endpoint.test.ts - Tests endpoint initialization

- endpoint-url-builder.ts + tests - Constructs endpoint URLs with environment support and validation

## Key Implementation
The follow up strategy extracts `answerId` from response headers and maintains it across the stream lifecycle, ensuring all dispatched actions correctly target the specific follow-up answer in Redux state.

[SFINT-6578]: https://coveord.atlassian.net/browse/SFINT-6578?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by erocheleau — APPROVED** [review]
Everything makes a lot of sense, I proposed some improvements, feel free to mark them with a TODO in the code and we can deal with them later, or implement a quick solution now. Up to you.

File: packages/headless/src/api/knowledge/answer-generation/endpoints/follow-up/url-builders/endpoint-url-builder.test.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+    it('should build the correct endpoint URL with valid state', () => {
+      const result = buildFollowUpEndpointUrl(state);
+
+      expect(getOrganizationEndpoint).toHaveBeenCalledWith(
+        mockOrganizationId,
+        'prod'
+      );
+      expect(result).toBe(
+        `${mockPlatformEndpoint}/rest/organizations/${mockOrganizationId}/answer/v1/configs/${mockAnswerConfigurationId}/generate`
```
Comment (SimonMilord) [inline packages/headless/src/api/knowledge/answer-generation/endpoints/follow-up/url-builders/endpoint-url-builder.test.ts]: as etienne said dont forget to change those (I guess if you do the tests will fail it should be ok)
